gnomad/utils/annotations.py

```diff
@@ -1446,3 +1373,166 @@ def annotate_downsamplings(
     )
 
     return t
+
+
+def compute_freq_by_strata(
+    mt: hl.MatrixTable,
+    strata_expr: List[Dict[str, hl.expr.StringExpression]],
+    downsamplings: Optional[List[int]] = None,
+    ds_pop_counts: Optional[Dict[str, int]] = None,
+    entry_agg_funcs: Optional[Dict[str, Tuple[Callable, Callable]]] = None,
+) -> hl.Table:
+    """
+    Compute call statistics and, when passed, entry aggregation function(s) by strata.
+
+    The computed call statistics are AC, AF, AN, and homozygote_count. Downsamplings are
+    added to the strata when downsamplings when passed. The entry aggregation functions
```

Comment:
I think somewhere, either here or in the other function we need to address the expectations for `pop_idx` and `global_idx` in `downsampling_expr` or any value in `strata_expr` with 'downsampling'  and it should reference `annotate_downsamplings`